src/main/java/com/redhat/devtools/lsp4ij/features/semanticTokens/viewProvider/LSPSemanticTokensFileViewProviderHelper.java

```diff
@@ -47,10 +51,39 @@ public LSPSemanticTokensFileViewProviderHelper(@NotNull LSPSemanticTokensFileVie
 
     @Nullable
     private PsiFile getFile() {
-        // There should only be one PSI file
-        List<PsiFile> allFiles = fileViewProvider.getAllFiles();
-        PsiFile file = allFiles.size() == 1 ? ContainerUtil.getFirstItem(allFiles) : null;
-        return (file != null) && file.isValid() && EditorBehaviorFeature.enableSemanticTokensFileViewProvider(file) ? file : null;
+        // Lazy-initialize the one PSI file for this view provider
```

Comment:
There are two aspects to this cache. First, the one file is cached on the file view provider helper instance via DCL lazy-initialization with automatic read-through when the cached file becomes invalid.